src/modules/hip/kernel/transpose.hpp

```diff
@@ -0,0 +1,101 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// Vectorized dst->src mapping
+template <typename T>
+__global__ void transpose_generic_hip_tensor(T *srcPtr,
+                                             uint *srcStrides,
+                                             T *dstPtr,
+                                             uint *dstStrides,
+                                             uint *dstDims,
+                                             uint dstNumDims,
+                                             uint *permTensor)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    if(id_x >= dstStrides[0])
+        return;
+
+    int maxLength = dstStrides[0];
+    int xAlignedLength =  maxLength & ~7;      // alignedLength for vectorized global loads
+    int xDiff = maxLength - xAlignedLength;    // difference between roiWidth and alignedLength
+    // Move dstIdx and srcIdx to start of given input tensor
+    uint dstIdx = (id_y * *dstStrides++);
+    uint srcIdx = (id_y * *srcStrides++);
+
+    d_uint8 dstCoords[RPPT_MAX_DIMS], srcIdxs;
+    uint4 idx0123 = make_uint4(id_x, id_x + 1, id_x + 2, id_x + 3);
+    uint4 idx4567 = make_uint4(id_x + 4, id_x + 5, id_x + 6, id_x + 7);
+    srcIdxs.ui4[0] = srcIdxs.ui4[1] = make_uint4(srcIdx, srcIdx, srcIdx, srcIdx);
+
+    // Compute 8 dstCoords given id_x
+    for (int i = 0; i < dstNumDims; i++)
+    {
+        dstCoords[i].ui4[0] = (idx0123 / dstStrides[i]) % dstDims[i];
+        dstCoords[i].ui4[1] = (idx4567 / dstStrides[i]) % dstDims[i];
+    }
+    // Compute corresponding 8 srcIdxs given id_x
+    for (int i = 0; i < dstNumDims; i++)
+    {
+        for (int j = 0; j < 8; j++)
+            srcIdxs.ui1[j] += (dstCoords[permTensor[i]].ui1[j] * srcStrides[permTensor[permTensor[i]]]);
```

Comment:
@snehaa8 or @sampath1117  This needs to be vectorized and the loop has to be gone:
8 uint muls -> 2 uint4 muls
8 uint adds -> 2 uint4 adds